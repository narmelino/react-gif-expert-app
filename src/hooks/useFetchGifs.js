import { useEffect, useState } from "react";
import getGifs from '../utils/getGifs';

const useFetchGifs = (category) => {
  const [state, setState] = useState({
    data: [],
    loanding: true,
  })

  useEffect(() => {
    getGifs(category).then(gifs => {
      setState({
        data: gifs,
        loanding: false,
      });
    });
  }, [category]);

  return state;
}

export default useFetchGifs;