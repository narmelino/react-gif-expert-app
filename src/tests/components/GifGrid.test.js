import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import useFetchGifs from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas con <GifGrid />', () => {
  //let wrapper = shallow(<GifGrid category="Goku" />);

  beforeEach(() => {
    
  });

  test('Debe renderizar el componente correctamente', () => {
    useFetchGifs.mockReturnValue({
      loanding: true,
      data: [],
    });
    const wrapper = shallow(<GifGrid category="Goku" />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe mostrar items cuando se cargan las imágenes', () => {
    const gifs = [{
      id: 'ABC',
      url: 'https://domain.com/image.png',
      title: 'Mock image',
    }];

    useFetchGifs.mockReturnValue({
      loanding: false,
      data: gifs,
    });
    const wrapper = shallow(<GifGrid category="Goku" />);
    expect(wrapper).toMatchSnapshot();

    // No debería estar el párrafo de cargando.
    expect(wrapper.find('p').exists()).toBe(false);

    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});