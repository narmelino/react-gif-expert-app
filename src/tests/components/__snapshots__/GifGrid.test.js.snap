// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con GifGrid Debe mostrar items cuando se cargan las im√°genes 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Goku
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Mock image"
      url="https://domain.com/image.png"
    />
  </div>
</Fragment>
`;

exports[`Pruebas con GifGrid Debe renderizar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Goku
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
